library/src/main/kotlin/kdocformatter/ParagraphListBuilder.kt

```diff
@@ -382,13 +382,18 @@ class ParagraphListBuilder(
         if (last.preformatted || last.doc || last.hanging && !last.continuation || last.isEmpty()) {
             return
         }
+
         val text = last.content
+        if (!text.startsWithUpperCaseLetter()) {
+            return
+        }
+
         for (i in text.length - 1 downTo 0) {
             val c = text[i]
             if (c.isWhitespace()) {
                 continue
             }
-            if (c.isLetterOrDigit() && text[0].isLetter() && text[0].isUpperCase()) {
+            if (c.isLetterOrDigit() || c.isCloseSquareBracket()) {
```

Comment:
Alternatively, we could also consider a blacklist approach, eg. `if (!c.isPunctuation())` - but I'm not sure what the reasoning behind this check should be. WDYT?